mathext/ell_complete.go

```diff
@@ -0,0 +1,157 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+)
+
+// CompleteK computes the complete elliptic integral of the 1st kind, 0≤m≤1:
+//
+//	K(m) = \int_{0}^{\pi/2} d\theta / {\sqrt{1-m{\sin^2\theta}}}
+//
+// See: http://dx.doi.org/10.1016/j.cam.2014.12.038 for the computation method.
+func CompleteK(m float64) float64 {
+	if !(0 <= m && m <= 1) {
```

Comment:
`if m < 0 || 1 < m {` and add a note in the doc comment that outside [0,1], `CompleteK` returns `math,NaN`.